src/hotspot/share/gc/g1/g1GCPhaseTimes.cpp

```diff
@@ -419,6 +420,8 @@ double G1GCPhaseTimes::print_pre_evacuate_collection_set() const {
 
   info_time("Pre Evacuate Collection Set", sum_ms);
 
+  // Concurrent tasks of ResetMarkingState and NoteStartOfMark are triggered during
+  // young collection. However, their execution time are not included in _gc_pause_time_ms.
   if (pre_concurrent_start_ms > 0.0) {
```

Comment:
Since `pre_concurrent_start_ms` is now actually gathered, maybe print an extra line for it, with the `ResetMarkingState` and `NoteStartOfMark` log lines indented?

I.e. something like:

```
  if (_cur_prepare_concurrent_task_time_ms > 0.0) {
    debug_time("Prepare Concurrent Start", _cur_prepare_concurrent_task_time_ms);
    debug_phase(_gc_par_phases[ResetMarkingState], 1);
    debug_phase(_gc_par_phases[NoteStartOfMark], 1);
  }
```
?

Then we can also drop the calculation of the local `pre_concurrent_start_ms`.

Reply:
Okay, this looks better.